tc/core/tensor.cc

```diff
@@ -107,6 +107,48 @@ bool TensorInfo::operator!=(const TensorInfo& t) const {
   return !(*this == t);
 }
 
+std::ostream& operator<<(std::ostream& os, const TensorInfo& t) {
+  auto ndim = t.shape.size();
+  if (ndim == 0) {
+    return os;
+  }
+  os << "Dimensions: ";
+  if (ndim) {
+    for (size_t i = 0; i < ndim; ++i) {
+      os << t.shape[i] << "  ";
+    }
+    os << std::endl;
+    os << "Strides: ";
+    if (t.strides.size()) {
+      for (size_t i = 0; i < ndim; ++i) {
+        os << t.strides[i] << "  ";
+      }
+    }
+  }
+  os << std::endl << "Type:";
```

Comment:
You don't want to force all these newlines, you want to give your caller the option to add a newline or not.
You want to print something like:
```
type: float base: 0x... offset: 0 sizes: [3, 4, 5] strides: [20 ,5, 1]
```